core/app/models/concerns/spree/user_reporting.rb

```diff
@@ -1,22 +1,28 @@
 module Spree
   module UserReporting
-    extend DisplayMoney
-    money_methods :lifetime_value, :average_order_value
+    def display_lifetime_value(store = nil, currency = nil)
+      store ||= Store.default
+      Spree::Money.new(lifetime_value(store), currency: currency || store.default_currency)
+    end
+
+    def lifetime_value(store = nil)
+      store ||= Store.default
+      store.orders.complete.sum(:total)
```

Comment:
```suggestion
      store ||= Store.default
      currency ||= store.default_currency
      store.orders.complete.where(currency: currency).sum(:total)
```

Reply:
I think we don't actually need the default currency in those methods. Orders alwyays have currency.
```
# core/app/models/spree/order.rb:717
    def ensure_currency_presence
      self.currency ||= store.default_currency || Spree::Config[:currency]
```

Reply:
I think we don't need default currency here. Stores always have at least one.
```
# backend/app/views/spree/admin/users/_lifetime_stats.html.erb:12
          <% current_store.supported_currencies.split(',').each do |currency| %>

# core/app/models/spree/store.rb:58
    before_save :ensure_supported_currencies, :ensure_supported_locales, :ensure_default_country

# core/app/models/spree/store.rb:160
    def ensure_supported_currencies
      return unless attributes.keys.include?('supported_currencies')
      return if supported_currencies.present?
      return if default_currency.blank?

      self.supported_currencies = default_currency
    end

```

Reply:
I think we don't need currency in those methods. They always have a currency.
```
# core/app/models/spree/store.rb
    with_options presence: true do
      validates :name, :url, :mail_from_address, :default_currency, :code
    end
...
    before_save :ensure_supported_currencies, :ensure_supported_locales, :ensure_default_country
...
    def ensure_supported_currencies
      return unless attributes.keys.include?('supported_currencies')
      return if supported_currencies.present?
      return if default_currency.blank?

      self.supported_currencies = default_currency
    end
```

Reply:
@tomurb we need to pass `current_currency` when user changes their currency from one of the supported currencies